### ISSUE TITLE
Detection of non-linear/non-convex problem

### ISSUE THREAD

**Issue by trulsf** [jump-dev/Xpress.jl#274]
The current implementation calls a different optimization routine depending on internal classification schemes 
```julia
    if model.has_nlp_constraints
        @checked Lib.XPRSnlpoptimize(model.inner, model.solve_method)
    elseif is_mip(model)
        @checked Lib.XPRSmipoptimize(model.inner, model.solve_method)
    else
        @checked Lib.XPRSlpoptimize(model.inner, model.solve_method)
    end
```
This does not seem to work properly for non-convex quadratic problems which are not set as having nonlinear constraints and thus failing when calling `XPRSlpoptimize`. Here is a small example that fails:
```julia
using Xpress

m = Model(Xpress.Optimizer)
@variable(m, x >= 0)
@variable(m, y >= 0)
@constraint(m, x + y + x*y <= 5)
@objective(m, Max, x + y)
optimize!(m)
```
Instead of relying on an internal classification scheme would it not be better to let the choice of algorithm be left for Xpress using the more general `XPRSoptimize` function? The type of optimization used can then be queried afterwards using the `XPRS_OPTIMIZETYPEUSED`-attribute if this is needed before solution retrieval.

**Comment by odow** [comment]
Is this on the latest tagged version or the latest `master`?

**Comment by odow** [comment]
Ah it's 
https://github.com/jump-dev/Xpress.jl/blob/60ae1dd4971cd5cda76d214deaf93aa4c12c3d95/src/MOI/MOI_wrapper.jl#L2910-L2916

**Comment by odow** [comment]
I think the issue was that 
https://github.com/jump-dev/Xpress.jl/blob/60ae1dd4971cd5cda76d214deaf93aa4c12c3d95/src/Lib/xprs.jl#L479-L482
and some users of Xpress.jl are still stuck on quite outdated versions of Xpress.

**Comment by trulsf** [comment]
That is a pretty old version of Xpress to support. But is it not possible to have code that checks the version of Xpress available and adjusts accordingly? With the current implementation we do not get access to the capabilities of the global solver for nonconvex quadratic problems (nor MIQCQPs where the quadratic part defines a non-convex region).
Alternatively, having some control mechanism to activate the global solver when calling `optimize!`.

**Comment by odow** [comment]
> But is it not possible to have code that checks the version of Xpress available and adjusts accordingly?

Yeah, we can probably do that. Especially now that we have CI for old and new versions.

> That is a pretty old version of Xpress to support

I won't name and shame the customer...

**Comment by GioniMexi** [comment]
Hi, I am facing the same issue for some problems with quadratic constraints. I am using a recent Xpress version (9.5) and #285 seems to fix the issue for me. I am wondering what is missing for #285 to be merged and why the CI is failing :)
